
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://github.com/WangLab-ComputationalBiology/btc-scrna-training/advanced/">
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../intro-cirro/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.2, mkdocs-material-9.2.8">
    
    
      
        <title>Advanced settings - BTC scRNA-Seq Material</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.046329b4.min.css">
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z"/></svg>');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
        html.glightbox-open { overflow: initial; height: 100%; }
        .gslide-title { margin-top: 0px; user-select: text; }
        .gslide-desc { color: #666; user-select: text; }
        .gslide-image img { background: white; }
        
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
            </style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
  
  
    <body dir="ltr">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#advanced-configuration" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="BTC scRNA-Seq Material" class="md-header__button md-logo" aria-label="BTC scRNA-Seq Material" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            BTC scRNA-Seq Material
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Advanced settings
            
          </span>
        </div>
      </div>
    </div>
    
    
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="BTC scRNA-Seq Material" class="md-nav__button md-logo" aria-label="BTC scRNA-Seq Material" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    BTC scRNA-Seq Material
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Getting started
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Getting started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Advanced settings
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Advanced settings
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-creating-nextflow-profiles" class="md-nav__link">
    1. Creating Nextflow profiles
  </a>
  
    <nav class="md-nav" aria-label="1. Creating Nextflow profiles">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-slurm-profile" class="md-nav__link">
    1.1 SLURM profile
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-container-and-nextflow-pipelines" class="md-nav__link">
    1.2. Container and Nextflow pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-pipeline-entrypoints" class="md-nav__link">
    2. Pipeline entrypoints
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-adding-gene-signatures-and-meta-programs" class="md-nav__link">
    3. Adding gene signatures and meta-programs
  </a>
  
    <nav class="md-nav" aria-label="3. Adding gene signatures and meta-programs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-cell-markers" class="md-nav__link">
    3.1. Cell Markers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-meta-programs" class="md-nav__link">
    3.2. Meta-programs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-adding-custom-genomes-on-the-pipeline-cellranger-alignment" class="md-nav__link">
    4. Adding custom genomes on the pipeline (Cellranger alignment).
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Documentation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Documentation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../intro-cirro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to Cirro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../command-line-interface/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Command-line interface
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Tutorial
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Tutorial
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../quality-control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    1. Sample and QC filtering
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../dimensionality-clustering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    2. Dimensionality reduction and clustering
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../stratification-and-cnv/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    3. Cell stratification and CNV Analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../batch-correction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    4. Batch correction and evaluation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../cell-annotation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    5. Cell annotation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../cell-communcation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    6. Cell-cell communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../meta-programs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    7. Malignant meta-programs
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../contributing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Contributing
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../faq/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FAQ
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    About
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            About
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../license/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    License
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../release-notes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Release Notes
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-creating-nextflow-profiles" class="md-nav__link">
    1. Creating Nextflow profiles
  </a>
  
    <nav class="md-nav" aria-label="1. Creating Nextflow profiles">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-slurm-profile" class="md-nav__link">
    1.1 SLURM profile
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-container-and-nextflow-pipelines" class="md-nav__link">
    1.2. Container and Nextflow pipelines
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-pipeline-entrypoints" class="md-nav__link">
    2. Pipeline entrypoints
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-adding-gene-signatures-and-meta-programs" class="md-nav__link">
    3. Adding gene signatures and meta-programs
  </a>
  
    <nav class="md-nav" aria-label="3. Adding gene signatures and meta-programs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-cell-markers" class="md-nav__link">
    3.1. Cell Markers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-meta-programs" class="md-nav__link">
    3.2. Meta-programs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-adding-custom-genomes-on-the-pipeline-cellranger-alignment" class="md-nav__link">
    4. Adding custom genomes on the pipeline (Cellranger alignment).
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="advanced-configuration">Advanced configuration<a class="headerlink" href="#advanced-configuration" title="Permanent link">#</a></h1>
<h3 id="1-creating-nextflow-profiles">1. Creating Nextflow profiles<a class="headerlink" href="#1-creating-nextflow-profiles" title="Permanent link">#</a></h3>
<p>For users running the pipeline in an HPC environment, it is necessary to set up a profile. Essentially, Nextflow <a href="https://www.nextflow.io/docs/latest/config.html#config-profiles">profiles</a> define specific settings related to the job scheduler of the HPC. Various HPCs may employ different engines for job scheduling, such as SLURM, TORQUE, and LSF. </p>
<h4 id="11-slurm-profile">1.1 SLURM profile<a class="headerlink" href="#11-slurm-profile" title="Permanent link">#</a></h4>
<p>The profiles should be written on the <code>nextflow.config</code> file. For HPCs, it is essential to load the <code>singularity</code> module. This instruction will replace the <code>module load</code> command.</p>
<div class="copy highlight"><pre><span></span><code>institution<span class="w"> </span><span class="o">{</span>

<span class="w">    </span><span class="nv">module</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">&#39;singularity/3.7.0&#39;</span>

<span class="w">    </span>singularity<span class="w"> </span><span class="o">{</span>

<span class="w">        </span><span class="nv">enabled</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="nb">true</span>

<span class="w">    </span><span class="o">}</span>

<span class="w">    </span>process<span class="w"> </span><span class="o">{</span>

<span class="w">        </span><span class="nv">executor</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s1">&#39;slurm&#39;</span>
<span class="w">        </span><span class="nv">queue</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s1">&#39;medium&#39;</span>

<span class="w">    </span><span class="o">}</span>

<span class="w">    </span>params<span class="w"> </span><span class="o">{</span>

<span class="w">        </span><span class="nv">max_memory</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="m">128</span>.GB
<span class="w">        </span><span class="nv">max_cpus</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="m">32</span>
<span class="w">        </span><span class="nv">max_time</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="m">24</span>.h

<span class="w">    </span><span class="o">}</span>
<span class="o">}</span>
</code></pre></div>
<p>The previous snippet creates a profile for a institution powered by SLURM. It includes several settings, such as loading modules like <code>singularity/3.7.0</code>, and specifying the job scheduling engine with the <code>executor</code> property. Nextflow works with many engines; for more details, refer to the official <a href="https://nf-co.re/docs/usage/tutorials/step_by_step_institutional_profile">documentation</a>. Please note that the <code>process</code> scope should match the HPC specs, i.e., queue, memory and cpus.</p>
<h4 id="12-container-and-nextflow-pipelines">1.2. Container and Nextflow pipelines<a class="headerlink" href="#12-container-and-nextflow-pipelines" title="Permanent link">#</a></h4>
<p>Why use Singularity/Docker? It's important to note that Nextflow pipelines integrate with containers to ensure reproducibility and portability. This integration guarantees that, during pipeline execution, each module operates within a predefined computational environment, which includes all required software, libraries, and programming languages. Consequently, researchers can trust that their workflows will yield consistent results across various platforms. Additionally, these workflows can be effortlessly shared and run by collaborators in different computing environments.</p>
<p><a class="glightbox" href="../figures/schema-container.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img align="center" alt="Image caption" src="../figures/schema-container.png" /></a></p>
<hr />
<h3 id="2-pipeline-entrypoints">2. Pipeline entrypoints<a class="headerlink" href="#2-pipeline-entrypoints" title="Permanent link">#</a></h3>
<p>This pipeline was designed with multiple entry points in mind. It is still a work in progress, but the pipeline already offers ways to skip a few modules (processes) and entire routines (sub-workflows). The entry points are named based on major functions: Basic, Stratification, Annotation, nonMalignant, and Malignant. Details are provided below.</p>
<p><a class="glightbox" href="../figures/schema-pipeline-entrypoint.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img align="center" alt="Image caption" src="../figures/schema-pipeline-entrypoint.png" /></a></p>
<hr />
<h3 id="3-adding-gene-signatures-and-meta-programs">3. Adding gene signatures and meta-programs<a class="headerlink" href="#3-adding-gene-signatures-and-meta-programs" title="Permanent link">#</a></h3>
<p>The pipeline allows flexibility regarding cell or meta-program markers. Currently, we provide two databases with curated cell markers <a href="https://github.com/break-through-cancer/btc-scrna-pipeline/blob/main/assets/cell_markers_database.csv">[1]</a> and meta-programs <a href="https://github.com/break-through-cancer/btc-scrna-pipeline/blob/main/assets/meta_programs_database.csv">[2]</a>.</p>
<h4 id="31-cell-markers">3.1. Cell Markers<a class="headerlink" href="#31-cell-markers" title="Permanent link">#</a></h4>
<p>To customize the cell markers database, it's essential to adhere to pipeline conventions. Firstly, the table should consist of four columns, as detailed below. Most importantly, the cell type column will determine the annotation level at which the pipeline will operate.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">cell_subset</th>
<th style="text-align: left;">cell_type</th>
<th style="text-align: left;">annotation</th>
<th style="text-align: left;">markers</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">T-Cells</td>
<td style="text-align: left;">CD3D</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">T-Cells</td>
<td style="text-align: left;">CD3E</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">T-Cells</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">T-Cells</td>
<td style="text-align: left;">CD8A</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">T-Cells</td>
<td style="text-align: left;">CD8B</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">NK Cells</td>
<td style="text-align: left;">NCAM1</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">NK Cells</td>
<td style="text-align: left;">KLRG1</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">NK Cells</td>
<td style="text-align: left;">FCGR3A</td>
</tr>
<tr>
<td style="text-align: left;">Lineage markers Subsets</td>
<td style="text-align: left;">Major cells</td>
<td style="text-align: left;">NK Cells</td>
<td style="text-align: left;">NKG7</td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Note that each marker (gene) is presented in a single row. This long format aligns with best practices for data analysis.</p>
</div>
<h4 id="32-meta-programs">3.2. Meta-programs<a class="headerlink" href="#32-meta-programs" title="Permanent link">#</a></h4>
<p>Similar to customizing cell markers, users will need to adhere to pre-established standards. The <code>source</code> column will serve as the anchor that the pipeline uses to subset the meta-programs database.</p>
<table>
<thead>
<tr>
<th style="text-align: left;">source</th>
<th style="text-align: left;">meta_program</th>
<th style="text-align: left;">gene_marker</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">SPARCL1</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">GFAP</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">CLU</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">CRYAB</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">TTYH1</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">SLC1A3</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">CST3</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">ID3</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">AGT</td>
</tr>
<tr>
<td style="text-align: left;">Malignant</td>
<td style="text-align: left;">Astrocytes</td>
<td style="text-align: left;">APOE</td>
</tr>
</tbody>
</table>
<hr />
<h3 id="4-adding-custom-genomes-on-the-pipeline-cellranger-alignment">4. Adding custom genomes on the pipeline (Cellranger alignment).<a class="headerlink" href="#4-adding-custom-genomes-on-the-pipeline-cellranger-alignment" title="Permanent link">#</a></h3>
<p>The pipeline employs Cellranger for alignment. We set up the reference genome based on Gencode (v46) and GRCH38. To achieve this, we adhered to the tutorial provided in the 10x Genomics <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#grch38_%23%7Bfiles.refdata_GRCh38.version%7D">documentation</a>. </p>
<p>Alternatively, users can substitute it with a version of their choice, but certain conventions must be observed. The <strong>mkref</strong> output should be stored in a folder that adheres to the following structure: </p>
<div class="highlight"><pre><span></span><code>Genomes/Homo_sapiens/ANNOTATION/GENOME_VERSION/
</code></pre></div>
<p>The terms <strong>ANNOTATION</strong> and <strong>GENOME_VERSION</strong> should be replaced with the user's preferred choices. Next, the user will need to edit a few lines on <code>conf/igenomes.config</code> within <code>btc-scrna-pipeline</code> folder. </p>
<div class="highlight"><pre><span></span><code>params {
    // illumina iGenomes reference file paths
    genomes {
        &#39;GRCh38&#39; {
            cellranger       = &quot;${params.igenomes_base}/Genomes/Homo_sapiens/Gencode46/GRCh38&quot;
        }
        // Add your custom genome here
        &#39;Custom&#39; {
            cellranger       = &quot;${params.igenomes_base}/Genomes/Homo_sapiens/ANNOTATION/GENOME_VERSION&quot;
        }
    }
}
</code></pre></div>
<p>Finally, the command line should have both <code>genome</code> and <code>igenomes_base</code> parameters.</p>
<div class="copy highlight"><pre><span></span><code>nextflow<span class="w"> </span>run<span class="w"> </span>single_cell_basic.nf<span class="w"> </span>--project_name<span class="w"> </span>Training<span class="w"> </span>--sample_csv<span class="w"> </span>sample_table.csv<span class="w"> </span>--meta_data<span class="w"> </span>meta_data.csv<span class="w"> </span>--cancer_type<span class="w"> </span>Ovarian<span class="w"> </span>--genome<span class="w"> </span>Custom<span class="w"> </span>path/to/Genomes/Homo_sapiens/ANNOTATION/GENOME_VERSION<span class="s2">&quot; --igenomes_base -resume -profile seadragon</span>
</code></pre></div>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.expand"], "search": "../assets/javascripts/workers/search.dfff1995.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.dff1b7c8.min.js"></script>
      
    
  <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>