# FAQs

## How can I add my gene signatures in the scRNA pipeline?

???+ Answer

    Yes. A user can easily add markers into [cell annotation](https://github.com/break-through-cancer/btc-scrna-pipeline/blob/main/assets/cell_markers_database.csv) database. Additionally, the user can contribute to cell annotation database by adding markers into the [collections](https://breakthroughcancer.sharepoint.com/:x:/r/sites/TeamLab-BreakThroughCancerInformation/Shared%20Documents/scRNA-seq/3.0%20Pipelines/scRNAseq%20pipeline%20required%20tools.xlsx?d=wa91ee12a14cb4fc49c0b8d2936c75300&csf=1&web=1&e=5cuDZa&nav=MTVfezZBMzNGQkYzLTYxQjQtNEQ5RS05Rjg1LUYwMUUzQkYxNTE1OX0) sheet.

## Can I use my custom reference genome on the pipeline?

???+ Answer

    ADD AN EXAMPLE - CARs, VIRUS, DISTINCT MODELS?
    Although not recommented for standards reasons. It is possible to add a custom genome at the Alignament step. Please, check the [Advanced](advanced.md) configurations for more explanation.

## Does the pipeline supports data generated by distinct protocols (e.g. scFPPE, snRNA-Seq, etc)?

???+ Answer

    Currently, the pipeline is designed to support libraries derived from fresh-frozen samples. However,  we will prioritize adapting the pipeline for FPPE-based samples and snRNA-Seq.

---

Got more questions? Please, send an email to **affaustino@mdanderson.org**. We're here to help!